packages/autoloader/src/functions.php

```diff
@@ -130,10 +129,9 @@ function reset_maps_after_update( $response, $hook_extra, $result ) {
 		$plugin_dir  = str_replace( '\\', '/', WP_PLUGIN_DIR );
 		$plugin_path = trailingslashit( $plugin_dir ) . trailingslashit( explode( '/', $plugin )[0] );
 
-		if ( is_readable( $plugin_path . 'vendor/autoload_functions.php' ) ) {
-			// The plugin has a v2.x autoloader, so reset it.
-			$jetpack_autoloader_latest_version = null;
-			$jetpack_packages_classmap         = array();
+		if ( is_readable( $plugin_path . 'vendor/jetpack-autoloader/autoload_functions.php' ) ) {
```

Comment:
I'm having a hard time understanding this change, do you mind explaining this one a bit more? So this only concerns cases where a consumer plugin updates to autoloader version 2.3, and is currently has the autoloader v2.3 loaded in memory, right? :exploding_head:

Reply:
### Stop the autoloaders that are already released from trying to find the latest installed autoloader during a plugin update.

In the code above, we changed the file path for `vendor/autoload_functions.php` to `vendor/jetpack-autoloader/autoload_functions.php`. The reason for change is to prevent the already released autoloaders from trying to find the latest installed autoloader. As you can [see here](https://github.com/Automattic/jetpack/blob/master/packages/autoloader/src/functions.php#L133), the already released autoloaders look for the `vendor/autoload_functions.php` to determine if the autoloader should be reset.

-----------------

You're right that this only concerns cases where a consumer plugin updates to autoloader >=v2.3, and the current autoloader in memory is >=v2.3.

This does mean that already released autoloaders will not reset the maps when a plugin update includes autoloader v2.3. The already released autoloaders are looking for `vendor/autoload_functions`, which doesn't exist in autoloader v2.3. As described above, we're intentionally trying to stop the released autoloaders from resetting after an update to prevent the error that could occur during an update.